apps/right-to-rent-check/behaviours/tenants.js

```diff
@@ -12,31 +13,106 @@ module.exports = fields => {
           return callback(err);
         }
 
-        // Find all legit fields
-        const picked = _.pick(values, fields);
-        // Get the list of tenants
-        const tenants = req.sessionModel.get('tenants') || [];
-        // Find a tenant matching the current tenant
-        let tenant = _.find(tenants, picked);
-
-        // Add a new tenant
-        if (!tenant) {
-          tenant = _.cloneDeep(picked);
-          if (!_.isEmpty(tenant)) {
-            tenants.push(tenant);
+        let tenant = _.pick(values, fields);
+        let tenants = req.sessionModel.get('tenants') || [];
+
+        const savedByUuid = _.find(tenants, {
+          'tenant-uuid': values['tenant-uuid']
+        });
+
+        // Have we just finished editting a tenant record
+        // or is our savedByUuid different from the current tenant
+        const editted = _.find(tenants, {edit: true}) ||
+          (!_.isEqual(_.omit(savedByUuid, 'edit', 'tenant-uuid'), tenant) ? savedByUuid : false);
```

Comment:
This is cool and works, it's just very hard to understand.  It would be good if this could be broken up a bit more